pkg/postgres/run.go

```diff
@@ -0,0 +1,48 @@
+package postgres
+
+import (
+	"strings"
+
+	"github.com/bipol/scrapedumper/pkg/martaapi"
+)
+
+type Run struct {
+	Identifier            string `json:"identifier"`
+	RunGroupIdentifier    string `json:"run_group_identifier"`
+	CorrectedLine         string `json:"corrected_line"`
+	CorrectedDirection    string `json:"corrected_direction"`
+	MostRecentEventMoment string `json:"most_recent_event_moment"`
+	RunFirstEventMoment   string `json:"run_first_event_moment"`
+
+	Line      martaapi.Line      `json:"line"`
+	Direction martaapi.Direction `json:"direction"`
+	TrainID   string             `json:"train_id"`
+
+	Arrivals Arrivals `json:"arrivals"`
+}
+
+func (r Run) setLineDirectionAndTrainID() {
+	parts := strings.Split(r.Identifier, "_")
+
+	r.Line = martaapi.Line(parts[0])
+	r.Direction = martaapi.Direction(parts[1])
+	r.TrainID = parts[2]
+}
+
+func (r Run) Finished() bool {
```

Comment:
This can be used by the websocket to interpret the changes and decide when a run can be removed from the state. Once an arrival time is set for the terminus of a line, no further updates will be made to the record, so it won't come up again in future polls of `GetRecentlyActiveRuns`